File: src/main/java/com/codingforcookies/betterrecords/gui/ContainerRecordEtcher.java
package com.codingforcookies.betterrecords.gui;

import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.entity.player.InventoryPlayer;
import net.minecraft.inventory.Container;
import net.minecraft.inventory.Slot;
import net.minecraft.item.ItemStack;

import com.codingforcookies.betterrecords.items.TileEntityRecordEtcher;

public class ContainerRecordEtcher extends Container {
	protected TileEntityRecordEtcher tileEntity;
	
	public ContainerRecordEtcher(InventoryPlayer inventoryPlayer, TileEntityRecordEtcher te){
		tileEntity = te;
		
		addSlotToContainer(new SlotValid(tileEntity, 0, 17, 50));
		
		for(int i = 0; i < 3; i++)
			for(int j = 0; j < 9; j++)
				addSlotToContainer(new Slot(inventoryPlayer, j + i * 9 + 9, 8 + j * 18, 84 + i * 18));
		
		for(int i = 0; i < 9; i++)
			addSlotToContainer(new Slot(inventoryPlayer, i, 8 + i * 18, 142));
	}
	
	public boolean canInteractWith(EntityPlayer player) {
		return tileEntity.isUseableByPlayer(player);
	}
	
	public ItemStack transferStackInSlot(EntityPlayer player, int slot) {
		return null;
	}
}

File: src/main/kotlin/com/codingforcookies/betterrecords/client/gui/ContainerRecordEtcher.kt
package com.codingforcookies.betterrecords.client.gui

import com.codingforcookies.betterrecords.block.tile.TileRecordEtcher
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.entity.player.InventoryPlayer
import net.minecraft.inventory.Container
import net.minecraft.inventory.Slot
import net.minecraft.item.ItemStack

class ContainerRecordEtcher(inventoryPlayer: InventoryPlayer, var tileEntity: TileRecordEtcher) : Container() {

    init {
        addSlotToContainer(SlotValid(tileEntity, 0, 17, 50))

        for (i in 0..2) {
            for (j in 0..8) {
                addSlotToContainer(Slot(inventoryPlayer, j + i * 9 + 9, 8 + j * 18, 84 + i * 18))
            }
        }

        for (i in 0..8) {
            addSlotToContainer(Slot(inventoryPlayer, i, 8 + i * 18, 142))
        }
    }

    override fun canInteractWith(player: EntityPlayer) =
            tileEntity.isUsableByPlayer(player)

    override fun transferStackInSlot(player: EntityPlayer?, slot: Int) =
            null
}
